randovania/games/prime_hunters/layout/force_field_configuration.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+import copy
+import dataclasses
+from enum import StrEnum
+from typing import TYPE_CHECKING, Self
+
+from randovania.bitpacking import bitpacking
+from randovania.bitpacking.bitpacking import BitPackDecoder, BitPackEnum, BitPackValue
+from randovania.game.game_enum import RandovaniaGame
+from randovania.game_description.db.configurable_node import ConfigurableNode
+from randovania.game_description.db.node_identifier import NodeIdentifier
+from randovania.lib import enum_lib
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+
+class LayoutForceFieldRequirement(BitPackEnum, StrEnum):
+    POWER_BEAM = "power-beam"
+    VOLT_DRIVER = "volt-driver"
+    MISSILE = "missile"
+    BATTLEHAMMER = "battlehammer"
+    IMPERIALIST = "imperialist"
+    JUDICATOR = "judicator"
+    MAGMAUL = "magmaul"
+    SHOCK_COIL = "shock-coil"
+    RANDOM = "random"
+
+    @classmethod
+    def from_item_short_name(cls, name: str) -> LayoutForceFieldRequirement:
+        for key, value in ITEM_NAMES.items():
+            if value == name:
+                return key
+        raise KeyError(f"Unknown name: {name}")
+
+    @property
+    def item_name(self) -> str:
+        if self == LayoutForceFieldRequirement.RANDOM:
+            raise ValueError("The random Requirement shouldn't be used for item_index")
+        return ITEM_NAMES[self]
+
+
+ITEM_NAMES = {
+    LayoutForceFieldRequirement.POWER_BEAM: "PowerBeam",
+    LayoutForceFieldRequirement.VOLT_DRIVER: "VoltDriver",
+    LayoutForceFieldRequirement.MISSILE: "Missile",
+    LayoutForceFieldRequirement.BATTLEHAMMER: "Battlehammer",
+    LayoutForceFieldRequirement.IMPERIALIST: "Imperialist",
+    LayoutForceFieldRequirement.JUDICATOR: "Judicator",
+    LayoutForceFieldRequirement.MAGMAUL: "Magmaul",
+    LayoutForceFieldRequirement.SHOCK_COIL: "ShockCoil",
+}
+enum_lib.add_long_name(
+    LayoutForceFieldRequirement,
+    {
+        LayoutForceFieldRequirement.POWER_BEAM: "Power Beam",
+        LayoutForceFieldRequirement.VOLT_DRIVER: "Volt Driver",
+        LayoutForceFieldRequirement.MISSILE: "Missile",
+        LayoutForceFieldRequirement.BATTLEHAMMER: "Battlehammer",
+        LayoutForceFieldRequirement.IMPERIALIST: "Imperialist",
+        LayoutForceFieldRequirement.JUDICATOR: "Judicator",
+        LayoutForceFieldRequirement.MAGMAUL: "Magmaul",
+        LayoutForceFieldRequirement.SHOCK_COIL: "Shock Coil",
+        LayoutForceFieldRequirement.RANDOM: "Random",
+    },
+)
+
+
+def _get_vanilla_force_field_configuration() -> dict[NodeIdentifier, LayoutForceFieldRequirement]:
+    from randovania.game_description import default_database
+
+    game = default_database.game_description_for(RandovaniaGame.METROID_PRIME_HUNTERS)
+    return {
+        node.identifier: LayoutForceFieldRequirement.from_item_short_name(node.extra["entity_type_data"]["weapon"])
```

Comment:
I wonder if it's possible to just check for the `type` field from extra instead of creating a new field that says the same thing but as a string.